selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/SeleniumWebDriverHelper.java

```diff
@@ -12,69 +12,461 @@
+    waitAndClickOnElement(elementLocator, DEFAULT_TIMEOUT);
+  }
+
+  /**
+   * Waits visibility of provided {@link WebElement} and click once on it by {@link
+   * WebElement#click()}
+   *
+   * @param webElement
+   * @param timeout waiting time in seconds
+   */
+  public void waitAndClickOnElement(WebElement webElement, int timeout) {
+    webDriverWaitFactory.get(timeout).until(visibilityOf(webElement)).click();
   }
 
+  /**
+   * Waits visibility of provided {@link WebElement} and click once on it by {@link
+   * WebElement#click()}
+   *
+   * @param webElement
+   */
   public void waitAndClickOnElement(WebElement webElement) {
-    loadPageWait.until(visibilityOf(webElement)).click();
+    waitAndClickOnElement(webElement, DEFAULT_TIMEOUT);
+  }
+
+  /**
+   * Moves cursor to {@link WebElement} with provided locator {@link By} and click twice on it by
+   * {@link org.openqa.selenium.interactions.Action} Used exactly {@link Actions#doubleClick()} and
+   * not {@link Actions#doubleClick(WebElement)} , it allows avoid {@link
+   * org.openqa.selenium.StaleElementReferenceException}
+   *
+   * @param elementLocator
+   */
+  public void moveToElementAndDoDoubleClick(By elementLocator) {
+    moveToElemet(elementLocator);
+    waitElementIsVisible(elementLocator);
+    actionsFactory.createAction(seleniumWebDriver).doubleClick().perform();
+  }
+
+  /**
+   * Moves cursor to provided {@link WebElement} and click twice on it by {@link
+   * org.openqa.selenium.interactions.Action} Used exactly {@link Actions#doubleClick()} and not
+   * {@link Actions#doubleClick(WebElement)} , it allows avoid {@link
+   * org.openqa.selenium.StaleElementReferenceException}
+   *
+   * @param webElement
+   */
+  public void moveToElementAndDoDoubleClick(WebElement webElement) {
+    moveToElemet(webElement);
+    waitElementIsVisible(webElement);
+    actionsFactory.createAction(seleniumWebDriver).doubleClick().perform();
   }
 
+  /**
+   * Moves cursor to {@link WebElement} with provided locator {@link By} and click once on it by
+   * {@link org.openqa.selenium.interactions.Action}
+   *
+   * @param elementLocator
+   */
+  public void moveToElementAndClickOnIt(By elementLocator) {
+    moveToElemet(elementLocator);
+    waitElementIsVisible(elementLocator);
+    actionsFactory.createAction(seleniumWebDriver).click().perform();
+  }
+
+  /**
+   * Moves cursor to provided {@link WebElement} and click once on it by {@link
+   * org.openqa.selenium.interactions.Action}
+   *
+   * @param webElement
+   */
+  public void moveToElementAndClickOnIt(WebElement webElement) {
+    moveToElemet(webElement);
+    waitElementIsVisible(webElement);
+    actionsFactory.createAction(seleniumWebDriver).click().perform();
+  }
+
+  /**
+   * Moves cursor to WebElement with provided locator which attached to DOM but it does not mean
+   * that element is visible
+   *
+   * @param elementLocator {@link By}
+   */
+  public void moveToElemet(By elementLocator) {
```

Comment:
`moveToElemet` > `moveCursorToElement`

Reply:
hoverToElement

Reply:
I would prefer `moveCursorToElement` so as it uses `action#moveTo()` method internally.